test/lifecycle_test.dart

```diff
@@ -145,11 +147,51 @@ main() {
         expect(
             component.lifecycleCalls,
             equals([
-              matchCall('componentWillUpdate', args: [expectedProps, initialState], state: initialState),
               matchCall('render', state: initialState),
-              matchCall('componentDidUpdate', args: [expectedProps, initialState], state: initialState),
+              matchCall('getSnapshotBeforeUpdate', args: [expectedProps, initialState], state: initialState),
+              matchCall('componentDidUpdate', args: [expectedProps, initialState, null], state: initialState),
             ].where((matcher) => matcher != null).toList()));
       });
+
+      test(
+          'getSnapshotBeforeUpdate returns the same value being passed into '
+          'componentDidUpdate', () {
+        LifecycleTestHelper component = getDartComponent(
+            render(components2.LifecycleTest({'getSnapshotBeforeUpdate': (_, __, ___) => 'snapshot'})));
+
+        component.lifecycleCalls.clear();
+        component.setState({});
+
+        expect(
+            component.lifecycleCalls,
+            containsAllInOrder([
+              matchCall('getSnapshotBeforeUpdate'),
+              matchCall('componentDidUpdate', args: [anything, anything, 'snapshot']),
+            ]));
+      });
+
+      test(
```

Comment:
I actually modified gregs jsBackedMaps type tests into a shared test in my pr that i use to check lots of types for context and in gregs jsBackedMaps tests that i think this would be beneficial to the use for testing the snapshot types here too.
Add this file: https://github.com/cleandart/react-dart/blob/4044e5c2e48224cd379d93c984665b78e149a179/test/shared_type_tester.dart

Then import it in this file:
`import 'shared_type_tester.dart';`

and the test would change to be like:
```
      group(
          'componentDidUpdate receives the same value created in getSnapshotBeforeUpdate',
          () {
        testSnapshotType(expectedSnapshot) {
          LifecycleTestHelper component = getDartComponent(
              render(components2.LifecycleTest({'getSnapshotBeforeUpdate': (_, __, ___) => expectedSnapshot})));

          component.lifecycleCalls.clear();
          component.setState({});

          expect(
            component.lifecycleCalls,
            containsAllInOrder([
              matchCall('getSnapshotBeforeUpdate'),
              matchCall('componentDidUpdate', args: [anything, anything, same(expectedSnapshot)])
            ]),
          );
        }

        sharedTypeTests(testSnapshotType);
      });
```

then if you do this you probably don't even need the test above.

Reply:
You can ignore this, ill add it to my pr